lib/gis_robot_suite/descriptive_metadata_builder.rb

```diff
@@ -0,0 +1,595 @@
+        "#{begin_date}-#{end_date}"
+      else
+        Date.parse(node.xpath('gml:TimeInstant/gml:timePosition', NS).text).strftime('%Y')
+      end
+    end
+
+    def coordinates_subjects
+      # use coordinates in native projection only, no longer reproject
+      extent = data_id_node.xpath('gmd:extent/gmd:EX_Extent/gmd:geographicElement/gmd:EX_GeographicBoundingBox', NS)
+      return unless extent.any?
+
+      { value: coordinates(extent),
+        type: 'map coordinates' } # also add field with the reference system
+    end
+
+    def coordinates(extent)
+      west = extent.xpath('gmd:westBoundLongitude/gco:Decimal', NS).text
+      east = extent.xpath('gmd:eastBoundLongitude/gco:Decimal', NS).text
+      north = extent.xpath('gmd:northBoundLatitude/gco:Decimal', NS).text
+      south = extent.xpath('gmd:southBoundLatitude/gco:Decimal', NS).text
+
+      values = [west, east, north, south].map(&:to_f)
+      to_coordinates_ddmmss(values)
+    end
+
+    def identifier
+      return [] unless data_id_node.xpath('gmd:citation/gmd:CI_Citation/gmd:identifier/gmd:MD_Identifier/gmd:code/gco:CharacterString[contains(text(), "doi")]', NS).any?
+
+      [{ displayLabel: 'DOI',
+         source: { code: 'doi' },
+         value: data_id_node.xpath('gmd:citation/gmd:CI_Citation/gmd:identifier/gmd:MD_Identifier/gmd:code/gco:CharacterString', NS).text }]
+    end
+
+    def purl
+      @purl ||= Settings.purl.url + "/#{bare_druid}"
+    end
+
+    def related_resource
+      [].tap do |resources|
+        resources << scanned_maps
+      end.compact.flatten
+    end
+
+    def scanned_maps
+      maps = iso19139_ng.xpath('//gmd:MD_Metadata/gmd:dataQualityInfo/gmd:DQ_DataQuality/gmd:lineage/gmd:LI_Lineage/gmd:source/gmd:LI_Source', NS)
+
+      return unless maps.xpath('gmd:description/gco:CharacterString[contains(text(), "Scanned map") or contains(text(), "Digitized map")]', NS).any?
+
+      maps.map do |related_map|
+        { type: 'has other format',
+          displayLabel: 'Scanned map',
+          title: [{ value: related_map.xpath('gmd:sourceCitation/gmd:CI_Citation/gmd:title/gco:CharacterString', NS).text }],
+          purl: related_map.xpath('gmd:sourceCitation/gmd:CI_Citation/gmd:identifier/gmd:MD_Identifier/gmd:code/gco:CharacterString', NS).text }
+      end
+    end
+
+    # rubocop:disable Layout/LineLength
+    def related_pubs_note
+      related_publications = data_id_node.xpath(
+        'gmd:aggregationInfo/gmd:MD_AggregateInformation/gmd:aggregateDataSetName/gmd:CI_Citation/gmd:otherCitationDetails/gco:CharacterString[contains(text(), "Related publication")]', NS
+      )
+      return unless related_publications.any?
+
+      related_publications.map do |node|
+        { type: 'citation/reference',
+          displayLabel: 'Related publication',
+          value: related_citation(node) }
+      end
+    end
+    # rubocop:enable Layout/LineLength
+
+    def related_citation(node)
```

Comment:
Creating a relatedResource with a citation for a related publication is new. This code creates a basic citation and may need to be updated in the future if other use cases emerge.